app/assets/javascripts/components/ChallengeModal/ChallengePrompt.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  ChallengePrompt,
+  ChallengeValidation,
+  ProtectionSessionDurations,
+} from '@standardnotes/snjs';
+import { FunctionComponent } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import { DecoratedInput } from '../DecoratedInput';
+import { DecoratedPasswordInput } from '../DecoratedPasswordInput';
+import { ChallengeModalValues } from './ChallengeModal';
+
+type Props = {
+  prompt: ChallengePrompt;
+  values: ChallengeModalValues;
+  index: number;
+  onValueChange: (value: string | number, prompt: ChallengePrompt) => void;
+  isInvalid: boolean;
+};
+
+export const ChallengeModalPrompt: FunctionComponent<Props> = ({
+  prompt,
+  values,
+  index,
+  onValueChange,
+  isInvalid,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (index === 0) {
+      inputRef.current?.focus();
+    }
+  }, [index]);
+
+  useEffect(() => {
+    if (isInvalid) {
+      inputRef.current?.focus();
+    }
+  }, [isInvalid]);
+
+  return (
+    <>
+      {prompt.validation === ChallengeValidation.ProtectionSessionDuration ? (
+        <>
```

Comment:
I guess the top-level fragment is sufficient and we can skip fragments in this and below conditions.
```suggestion
```